### PR TITLE
Allow deriving Parse on keywords.

### PR DESCRIPTION
**Pull request by emilio**
This makes patches like #19576 much easier.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19578)

### CODE REVIEW

**Comment by emilio** [comment]
(or anyone else)

**Comment by Manishearth** [comment]
✨ ✨ ✨ ✨ ✨ ✨ ✨ ✨ 💯💯 💯 💯 💯 💯 💯 💯 💯 💯 ✨ ✨ ✨ ✨ ✨ ✨ ✨ ✨

**Comment by emilio** [comment]
> ✨ ✨ ✨ ✨ ✨ ✨ ✨ ✨ 💯💯 💯 💯 💯 💯 💯 💯 💯 💯 ✨ ✨ ✨ ✨ ✨ ✨ ✨ ✨

Does this mean r=Manish? :P

**Comment by Manishearth** [comment]
lol no it means I haven't reviewed it yet (doing so)

feel free to land with r=✨  and break all the tools that deal with m-c commits written in Python 2.

**Comment by Manishearth** [comment]
dammit wrong button

**Comment by emilio** [comment]
lol

File: components/style_derive/lib.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#![recursion_limit = "128"]
```
Comment (Manishearth) [inline components/style_derive/lib.rs]: oh no

**Review by Manishearth — APPROVED** [review]
r+

File: components/style_derive/cg.rs [hunk 2]
```diff
@@ -406,3 +406,41 @@ pub fn where_predicate(
...
+/// If the first Camel segment is "Moz" or "Webkit", the result string
+/// is prepended with "-".
+pub fn to_css_identifier(mut camel_case: &str) -> String {
+    camel_case = camel_case.trim_right_matches('_');
+    let mut first = true;
+    let mut result = String::with_capacity(camel_case.len());
+    while let Some(segment) = split_camel_segment(&mut camel_case) {
+        if first {
+            match segment {
+                "Moz" | "Webkit" => first = false,
```
Comment (canaltinova) [inline components/style_derive/cg.rs]: We should also add `"Servo"` here.

Comment (emilio) [reply components/style_derive/cg.rs]: This is just moving code around, but yeah, you're right.

**Pull request merged**